### DISCUSSION TITLE
using mcu protocol on potentiometer

### DISCUSSION THREAD

**Discussion by benwadub** [tttapa/Control-Surface#703] [Q&A]
hi, i d like to know if that is possible to use the mcu protocol to automat the pan and send to my potentiometer, I know that normally we have to use it with encoders but it could be nice if there was a solution to map the pan and send on my potentiometers directly as I.  have 7 pots by track. anyone got suggestion to achieve it please?

**Comment by tttapa** [comment]
What do you mean by “sending to a potentiometer”? What would you send? How would the potentiometer react to it?

**Comment by benwadub** [comment]
I mean using one potentiometer for pan, one for aux1 send one for aux send2 instead to use only one encoder for all theses values.

**Reply by tttapa** [reply]
I'm sorry, I don't think I understand what you mean. But if you want to use a potentiometer (a variable resistor) to control the pan using MCU, the answer is no, the MCU protocol only allows absolute potentiometers to be used for the volume, not for pan or aux levels.

**Comment by benwadub** [comment]
Yes it s what I wanted, I d like to have access to all the mcu protocol parameters without to have to push button to scroll between pan, aux send…
That should be nice to have my controller working on every daw, I still can t find a way to output midi from logic to my leds and I can t find information about the lua script that logic use to do this.
Again thanks for your answer!

**Reply by Veroledez** [reply]
Hola, creo que para que no tengas que precionar  ningun boton debes escribir tu propio script, ya que el protocolo mackie activa los envios, pan, instr, plugins y demas cosas con botones específicos.

Tal vez puedas usar un potenciometro como pan en mackie, pero tendrias  que crear una clase en donde a partir de cada cambio del potenciometro envie un incremento o decremento, tendrias que añadir un pin analogico y quitar los 2 que define incrementdrecrementbuttons.

Es solo una idea, no lo he probado.

**Reply by benwadub** [reply]
hi, yes my problem is here it s that I can t find information on how to write script for Logic Pro,
logic has two kind of script apparently some are Lua scripts, theses ones are readable in a text editor I m looking for the moment at theses Lua script that are pre made for natively supported controllers but none have midi feedback. 
there are also others controllers scripts that are grouped in files as midi plugins and the files are some .bundle files and in theses files nothing seems to be modifiable
here is an exemple of theses files
[Logic Control.bundle.zip](https://github.com/tttapa/Control-Surface/files/8023979/Logic.Control.bundle.zip)

**Reply by Veroledez** [reply]
No uso logic, por lo que no tengo idea de como funciona, no se si hayas leido esto.

https://www.marcogalvan.com/2017/02/editor-de-scripts-de-logic-pro.html?m=1

Parece que tiene lo que buscas. O no se, lamento no poder ayudarte

**Reply by benwadub** [reply]
I know scripter, it s not exactly what I need, s rioter is something to manipulate midi over code inside logic, but it s not done to output midi from logic, but thanks for help!
